src/HospitalAPI/Controllers/Intranet/MapController.cs

```diff
@@ -39,23 +41,38 @@ public MapController(IMapService mapService, IRoomService roomService, IFloorSer
         public async Task<IActionResult> GetAllBuildings()
         {
             IEnumerable<MapBuilding> mapBuildings = await _mapService.GetAllBuildings();
-            return Ok(mapBuildings);
+            List<MapBuildingDto> mapBuildingDtos = new List<MapBuildingDto>();
+            foreach (MapBuilding mapBuilding in mapBuildings)
+            {
+                mapBuildingDtos.Add(new MapBuildingDto(mapBuilding));
+            }
+            return Ok(mapBuildingDtos.AsEnumerable());
         }
 
         [HttpGet]
         [Route("floors/{id}")]
         public async Task<IActionResult> GetFloorsByBuilding(int id)
         {
             IEnumerable<MapFloor> mapFloors = await _mapService.GetFloorsByBuilding(id);
-            return Ok(mapFloors);
+            List<MapFloorDto> mapFloorDtos = new List<MapFloorDto>();
```

Comment:
you can use Ienumerable from the start as here 
![image](https://user-images.githubusercontent.com/93371169/211168164-e8102830-2dbf-4e8b-80d9-1acd0e2c6adc.png)